### PR TITLE
fix(lpeg): Removing static build of lpeg

### PR DESCRIPTION
**Pull request by hurricanehrndz**
These  fixes should be contributed and fixed back to nixpkg master.

### CODE REVIEW

**Comment by GaetanLepage** [comment]
> These fixes should be contributed and fixed back to nixpkg master.

What I don't understand is that this logic seems to be present on the upstream derivation of neovim.

**Comment by hurricanehrndz** [comment]
Not all, read closely

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: flake/packages/neovim.nix [hunk 1]
```diff
@@ -99,12 +101,55 @@ in
...
+          }))
+        else luapkgs.lpeg;
+      requiredLuaPkgs = ps: (
+        with ps; [
+          (nvim-lpeg-dylib ps)
+          luabitop
+          mpack
+        ]
+      );
+      myNeovimLuaEnv = pkgs.luajit.withPackages requiredLuaPkgs;
```
Comment (willruggiano) [inline flake/packages/neovim.nix:141]: Looks good. One nit: remove this and just inline it below.

**Review by willruggiano — CHANGES_REQUESTED** [review]

File: flake/packages/neovim.nix (hunk 1 above)
Comment (willruggiano) [reply flake/packages/neovim.nix]: jk i got u

**Comment by GaetanLepage** [comment]
> These fixes should be contributed and fixed back to nixpkg master.

Can this be done today or should we delay it until nvim 0.11 lands on master ?

File: flake/packages/neovim.nix [hunk 2]
```diff
@@ -99,12 +99,53 @@ in
...
+        if pkgs.stdenv.hostPlatform.isDarwin
+        then
+          (luapkgs.lpeg.overrideAttrs (oa: {
+            preConfigure = ''
+              # neovim wants clang .dylib
+              sed -i makefile -e "s/CC = gcc/CC = clang/"
+              sed -i makefile -e "s/-bundle/-dynamiclib/"
+              sed -i makefile -e "s/lpeg.so/lpeg.dylib/"
+              sed -i makefile -e '/^linux:$/ {N; d;}'
+              cat makefile
```
Comment (GaetanLepage) [inline flake/packages/neovim.nix:113]:
```suggestion
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Ok I see ! Thanks for taking the time to fix it.
I will try to upstream it if possible/

**Comment by GaetanLepage** [comment]
As this patch [ended up being upstream](https://github.com/NixOS/nixpkgs/pull/351423), I will close this PR.
Thank you for submitting it in the first place !

**Pull request closed without merging**